backend/services/event.py

```diff
@@ -189,3 +233,296 @@ def delete(self, subject: User, id: int) -> None:
+        """
+        # Feature-specific authorization: User is getting their own registrations
+        # Administrative Permission: user.event_registrations : user/{user_id}
+        if subject.id != user.id:
+            self._permission.enforce(
+                subject,
+                "user.event_registrations",
+                f"user/{user.id}",
+            )
+
+        registration_entities = (
+            self._session.query(EventRegistrationEntity)
+            .where(EventRegistrationEntity.user_id == user.id)
+            .join(EventEntity, EventRegistrationEntity.event_id == EventEntity.id)
+            .where(EventEntity.time >= time_range.start)
+            .where(EventEntity.time < time_range.end)
+        ).all()
+
+        return [entity.to_model() for entity in registration_entities]
+
+    def get_event_registration_status(self, event_id: int) -> EventRegistrationStatus:
+        """
+        Retrieves the number of registrations for a given event.
+
+        Args:
+            event_id: a valid int representing the
+
+        Returns:
+            status: a valid EventRegistrationStatus representing the number of registrations for an event
+        """
+        count = (
+            self._session.query(EventRegistrationEntity)
+            .where(
+                EventRegistrationEntity.event_id == event_id,
+                EventRegistrationEntity.is_organizer == False,
+            )
+            .count()
+        )
+
+        status = EventRegistrationStatus(registration_count=count)
+        return status
+
+    def get_event_registration_statuses(
+        self, subject: User, time_range: TimeRange
+    ) -> list[UserRegistrationStatus]:
+        """
+        Get the registration status for all events a user is registered for in the time range
+
+        Args:
+            subject: The User making the request.
+            time_range: The period over which to search for event registrations.
+
+        Returns:
+            list[UserRegistrationStatus]: list of valid UserRegistrationStatus models representing whether or not a user is registered for each event
+        """
+        event_entities = self._session.query(EventEntity)
+
+        registrations_for_user = (
+            self._session.query(EventRegistrationEntity)
+            .where(EventRegistrationEntity.user_id == subject.id)
+            .join(EventEntity, EventRegistrationEntity.event_id == EventEntity.id)
+            .where(EventEntity.time >= time_range.start)
+            .where(EventEntity.time < time_range.end)
+        ).all()
+
+        events_registered_for = set(
+            [registration.event_id for registration in registrations_for_user]
+        )
+
+        return [
+            UserRegistrationStatus(
+                event_id=event.id, is_registered=event.id in events_registered_for
+            )
+            for event in event_entities
+        ]

```

Comment:
We will need to approach this differently. First, all events in the history of time are loaded on line 509 (and iterated over here). Second, I would argue that either this service method makes sense to return only a list of event models (perhaps without details) that a user is registered for, _or_ perhaps we have a variation on an `EventModel` where the field of `is_registered` is added to it, and there's an endpoint that a logged in user can query to get all events with registration details _as the user_. When not logged in, a different endpoint could be used for the public event listing page, and perhaps this just defaults to false for `is_registered` (or returns the more limited model). It his makes sense to talk through, happy to jump on a zoom!